File: src/APIBundle/Entity/Incident.php
<?php

namespace APIBundle\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="incidents")
 */
class Incident
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $incident_id;

    /**
     * @ORM\Column(type="string")
     */
    protected $incident_title;

    /**
     * @ORM\Column(type="text")
     */
    protected $incident_story;

    /**
     * @ORM\Column(type="integer")
     */
    protected $incident_weight;

    /**
     * @ORM\ManyToOne(targetEntity="IncidentType", inversedBy="type_incidents")
     * @ORM\JoinColumn(name="incident_type", referencedColumnName="type_id", onDelete="CASCADE")
     * @var IncidentType
     */
    protected $incident_type;

    /**
     * @ORM\ManyToOne(targetEntity="Event", inversedBy="event_incidents")
     * @ORM\JoinColumn(name="incident_event", referencedColumnName="event_id", onDelete="CASCADE")
     * @var Event
     */
    protected $incident_event;

    /**
     * @ORM\ManyToMany(targetEntity="Subobjective", inversedBy="incident_subobjectives")
     * @ORM\JoinTable(name="incidents_subobjectives",
     *      joinColumns={@ORM\JoinColumn(name="incident_id", referencedColumnName="incident_id", onDelete="CASCADE")},
     *      inverseJoinColumns={@ORM\JoinColumn(name="subobjective_id", referencedColumnName="subobjective_id", onDelete="CASCADE")}
     *      )
     * @var Subobjective[]
     */
    protected $incident_subobjectives;

    /**
     * @ORM\OneToOne(targetEntity="Outcome", mappedBy="outcome_incident")
     */
    protected $incident_outcome;

    /**
     * @ORM\OneToMany(targetEntity="Inject", mappedBy="inject_incident")
     * @var Inject[]
     */
    protected $incident_injects;

    /**
     * @ORM\Column(type="smallint")
     */
    protected $incident_order = 0;

    protected $incident_exercise;

    public function __construct()
    {
        $this->incident_objectives = new ArrayCollection();
        $this->incident_injects = new ArrayCollection();
    }

    public function getIncidentId()
    {
        return $this->incident_id;
    }

    public function setIncidentId($id)
    {
        $this->incident_id = $id;
        return $this;
    }

    public function getIncidentTitle()
    {
        return $this->incident_title;
    }

    public function setIncidentTitle($title)
    {
        $this->incident_title = $title;
        return $this;
    }

    public function getIncidentStory()
    {
        return $this->incident_story;
    }

    public function setIncidentStory($story)
    {
        $this->incident_story = $story;
        return $this;
    }

    public function getIncidentWeight()
    {
        return $this->incident_weight;
    }

    public function setIncidentWeight($weight)
    {
        $this->incident_weight = $weight;
        return $this;
    }

    public function getIncidentType()
    {
        return $this->incident_type;
    }

    public function setIncidentType($type)
    {
        $this->incident_type = $type;
        return $this;
    }

    public function getIncidentEvent()
    {
        return $this->incident_event;
    }

    public function setIncidentEvent($event)
    {
        $this->incident_event = $event;
        return $this;
    }

    public function getIncidentSubobjectives()
    {
        return $this->incident_subobjectives;
    }

    public function setIncidentSubobjectives($subobjectives)
    {
        $this->incident_subobjectives = $subobjectives;
        return $this;
    }

    public function getIncidentOutcome()
    {
        return $this->incident_outcome;
    }

    public function setIncidentOutcome($outcome)
    {
        $this->incident_outcome = $outcome;
        return $this;
    }

    public function getIncidentInjects()
    {
        return $this->incident_injects;
    }

    public function setIncidentInjects($injects)
    {
        $this->incident_injects = $injects;
        return $this;
    }

    public function getIncidentOrder()
    {
        return $this->incident_order;
    }

    public function setIncidentOrder($order)
    {
        $this->incident_order = $order;
        return $this;
    }

    public function getIncidentExercise()
    {
        return $this->incident_exercise;
    }

    public function setIncidentExercise($exercise)
    {
        $this->incident_exercise = $exercise;
        return $this;
    }
}

File: openex-player/src/main/java/io/openex/database/model/Incident.java
package io.openex.database.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.openex.helper.MonoModelDeserializer;
import io.openex.helper.MultiModelDeserializer;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "incidents")
public class Incident implements Base {
    @Id
    @Column(name = "incident_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("incident_id")
    private String id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "incident_type")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("incident_type")
    private IncidentType type;

    @ManyToOne
    @JoinColumn(name = "incident_event")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("incident_event")
    private Event event;

    @Column(name = "incident_title")
    @JsonProperty("incident_title")
    private String title;

    @Column(name = "incident_story")
    @JsonProperty("incident_story")
    private String story;

    @Column(name = "incident_weight")
    @JsonProperty("incident_weight")
    private Integer weight;

    @Column(name = "incident_order")
    @JsonProperty("incident_order")
    private Short order;

    @OneToMany(mappedBy = "incident", fetch = FetchType.EAGER)
    @Fetch(value = FetchMode.SUBSELECT)
    @JsonProperty("incident_injects")
    private List<Inject<?>> injects = new ArrayList<>();

    @OneToOne(mappedBy = "incident")
    @JsonProperty("incident_outcome")
    private Outcome outcome;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "incidents_subobjectives",
            joinColumns = @JoinColumn(name = "incident_id"),
            inverseJoinColumns = @JoinColumn(name = "subobjective_id"))
    @JsonSerialize(using = MultiModelDeserializer.class)
    @JsonProperty("incident_subobjectives")
    private List<SubObjective> subObjectives = new ArrayList<>();

    // region transient
    @JsonProperty("incident_exercise")
    public String getExerciseId() {
        return getEvent().getExercise().getId();
    }
    // endregion

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Event getEvent() {
        return event;
    }

    public void setEvent(Event event) {
        this.event = event;
    }

    public IncidentType getType() {
        return type;
    }

    public void setType(IncidentType type) {
        this.type = type;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getStory() {
        return story;
    }

    public void setStory(String story) {
        this.story = story;
    }

    public Integer getWeight() {
        return weight;
    }

    public void setWeight(Integer weight) {
        this.weight = weight;
    }

    public Short getOrder() {
        return order;
    }

    public void setOrder(Short order) {
        this.order = order;
    }

    public List<Inject<?>> getInjects() {
        return injects;
    }

    public void setInjects(List<Inject<?>> injects) {
        this.injects = injects;
    }

    public Outcome getOutcome() {
        return outcome;
    }

    public void setOutcome(Outcome outcome) {
        this.outcome = outcome;
    }

    public List<SubObjective> getSubObjectives() {
        return subObjectives;
    }

    public void setSubObjectives(List<SubObjective> subObjectives) {
        this.subObjectives = subObjectives;
    }
}
